litellm/integrations/keywordsai.py

```diff
@@ -0,0 +1,145 @@
+from litellm.utils import ModelResponse
+from litellm import stream_chunk_builder, print_verbose
+from litellm.integrations.custom_logger import CustomLogger
+import requests
+import os
+import json
+import datetime
+
+class BgColors:
+    HEADER = '\033[95m'
+    OKBLUE = '\033[94m'
+    OKCYAN = '\033[96m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+
+KEYWORDS_AI_API_KEY = os.getenv("KEYWORDS_AI_API_KEY")
+def setup_keywordsai_params(model, messages, kwargs):
+    url = "https://api.keywordsai.co/api/request-logs/create/"
+
+    if not KEYWORDS_AI_API_KEY:
+        print(f"{BgColors.WARNING}!!!!WARNING: KEYWORDS_AI_API_KEY not found in environment!!!!{BgColors.ENDC}")
+    headers = {
+        "Authorization": f"Bearer {KEYWORDS_AI_API_KEY}",
+        "Content-Type": "application/json",
+    }
+
+    keywordsai_params = kwargs.get("extra_body", {}).pop("keywordsai_params", {})
+    kwargs["keywordsai_url"] = url
+    kwargs["keywordsai_headers"] = headers
+    kwargs["keywordsai_stream_collector"] = []
+    kwargs["keywordsai_payload"] = {
+        "model": kwargs.get("model"),
+        "prompt_messages": kwargs.get("messages"),
+        "tool_choice": kwargs.get("tool_choice"),
+        "tools": kwargs.get("tools"),
+        "customer_identifier": keywordsai_params.get("customer_identifier", ""),
+        "thread_identifier": keywordsai_params.get("thread_identifier", ""),
+        "metadata": keywordsai_params.get("metadata"),
+        "stream": kwargs.get("stream", False),
+    }
+
+
+def modify_params(
+    kwargs, response_obj, start_time: datetime.datetime, end_time, status_code=200
+):
+    stream = kwargs.get("stream", False)
+    if stream:
+        if (
+            "time_to_first_token" not in kwargs["keywordsai_payload"]
+            and end_time is not None
+        ):
+            time_to_first_token = (end_time - start_time).total_seconds()
+            kwargs["keywordsai_payload"]["time_to_first_token"] = time_to_first_token
+
+        if isinstance(response_obj, ModelResponse):
+            if stream:
+                kwargs["keywordsai_stream_collector"].append(response_obj)
+            else:
+                kwargs["keywordsai_payload"][
+                    "full_response"
+                ] = response_obj.model_dump()
+    else:
+        original_response = kwargs.get("original_response")
+        if isinstance(original_response, str):
+            kwargs["keywordsai_payload"]["full_response"] = json.loads(
+                original_response
+            )
+        elif isinstance(original_response, ModelResponse):
+            kwargs["keywordsai_payload"][
+                "full_response"
+            ] = original_response.model_dump()
+
+
+def commit_to_keywordsai(kwargs, start_time, end_time, success=True):
```

Comment:
can you refactor this, to work like langsmith logging - specifically how batching logs works - https://github.com/BerriAI/litellm/blob/a3d4bf6c27cab69cb9ada11f3030657cdc16c5a4/litellm/integrations/langsmith.py#L364

Reply:
Resolved.

Sorry for the confusion, I meant to submit the collection of stream chunks. I used the existing complete_stream_response attribute from kwargs in my new pr